app/translators/tdr_manifest_to_rawls.py

```diff
@@ -47,16 +47,27 @@ def translate(self, import_details: Import, file_like: IO) -> Iterator[Entity]:
         TDRManifestToRawls.save_snapshot_id(import_details.id, source_snapshot_id)
 
         tables = parsed_manifest.get_tables()
-        return itertools.chain(*self.translate_tables(import_details, source_snapshot_id, tables))
+        return itertools.chain(*self.translate_tables(import_details, source_snapshot_id, tables, parsed_manifest.is_cyclical()))
 
     @classmethod
-    def translate_tables(cls, import_details: Import, source_snapshot_id: str, tables: List[TDRTable]) -> Iterator[Iterator[Entity]]:
+    def translate_tables(cls, import_details: Import, source_snapshot_id: str, tables: List[TDRTable], is_cyclical: bool) -> Iterator[Iterator[Entity]]:
         """Converts a list of TDR tables, each of which contain urls to parquet files, to an iterator of Entity objects."""
         pet_key = sam.admin_get_pet_key(import_details.workspace_google_project, import_details.submitter)
+        if not is_cyclical:
+            for t in tables:
+                for f in t.parquet_files:
+                    pt = ParquetTranslator(t, f, import_details, source_snapshot_id, pet_key, is_cyclical)
+                    yield pt.translate()
+        else:
+            yield from itertools.chain(TDRManifestToRawls.translate_ref_tables(import_details, source_snapshot_id, tables, True, pet_key, False), TDRManifestToRawls.translate_ref_tables(import_details, source_snapshot_id, tables, True, pet_key, True))
```

Comment:
I don't love that the `for t in tables:` loop is duplicated here and in `translate_ref_tables`. Could this method always delegate to `translate_ref_tables` and have that be the only place the loop is defined? I think the following would work, but I have not fully verified, please check my logic:

```suggestion
        if not is_cyclical:
            # when passing is_cyclical=False to translate_ref_tables, the value of translate_ref is irrelevant
            yield from itertools.chain(TDRManifestToRawls.translate_ref_tables(import_details, source_snapshot_id, tables, False, pet_key, False))
        else:
            yield from itertools.chain(TDRManifestToRawls.translate_ref_tables(import_details, source_snapshot_id, tables, True, pet_key, False), TDRManifestToRawls.translate_ref_tables(import_details, source_snapshot_id, tables, True, pet_key, True))
```